setup.py

```diff
@@ -3,48 +3,86 @@
 from setuptools import setup, find_packages
 
 
-setup(
-    name='py-evm',
-    # *IMPORTANT*: Don't manually change the version here. Use the 'bumpversion' utility.
-    version='0.2.0-alpha.14',
-    description='Python implementation of the Ethereum Virtual Machine',
-    long_description_markdown_filename='README.md',
-    author='Piper Merriam',
-    author_email='piper@pipermerriam.com',
-    url='https://github.com/ethereum/py-evm',
-    include_package_data=True,
-    py_modules=['evm', 'trinity', 'p2p'],
-    install_requires=[
+deps={
+    'evm': [
         "aiohttp>=2.3.1,<3.0.0",
         "async_lru>=0.1.0,<1.0.0",
+        "coincurve>=7.0.0,<8.0.0",
         "cryptography>=2.0.3,<3.0.0",
         "cytoolz>=0.9.0,<1.0.0",
         "eth-bloom>=1.0.0,<2.0.0",
+        "eth-hash[pycryptodome]",
         "eth-utils>=1.0.1,<2.0.0",
+        "plyvel==1.0.4",
         "pyethash>=0.1.27,<1.0.0",
         "py-ecc>=1.4.2,<2.0.0",
         "rlp>=0.4.7,<1.0.0",
         "eth-keys>=0.2.0b3,<1.0.0",
         "trie>=1.3.2,<2.0.0",
     ],
-    extras_require={
-        'coincurve': [
-            "coincurve>=7.0.0,<8.0.0",
-        ],
-        'leveldb': [
-            "plyvel==1.0.4",
-        ],
-        'trinity': [
-            "plyvel==1.0.4",
-            "coincurve>=7.0.0,<8.0.0",
-            "web3>=4.0.0b11,<5.0.0",
-        ],
-        'p2p': [
+    'p2p': [
             "aiohttp>=2.3.1,<3.0.0",
             "async_lru>=0.1.0,<1.0.0",
             "pysha3>=1.0.0,<2.0.0",
-        ],
-    },
+    ],
+    'trinity': [
+            "plyvel==1.0.4",
+            "coincurve>=7.0.0,<8.0.0",
+            "web3>=4.0.0b11,<5.0.0",
+    ],
+    'test': [
+        "flake8==3.5.0",
+        "mypy>=0.580",
```

Comment:
this should probably be under a separate `lint` section so that in tox we can reduce the installation size for the linting and mypy runs

Reply:
We can do that but considering that we cache all packages between runs anyway I wouldn't expect it to have a big impact on CI performance.

Reply:
Less about performance and more about isolation.  Not a big deal but it matches the convention we've used elsewhere and I think I like the distinction.